CriteoPublisherSdk/Sources/Logging/CR_Logging.h

```diff
@@ -35,6 +35,8 @@ NS_ASSUME_NONNULL_BEGIN
 
 @interface CR_Logging : NSObject <CR_LogHandler>
 
+- (instancetype)initWithLogHandler:(id<CR_LogHandler>)handler;
+
 + (void)logMessage:(CR_LogMessage *)message;
 + (instancetype)sharedInstance;
 + (void)setSharedLogHandler:(id<CR_LogHandler>)handler;
```

Comment:
Indeed the setter is not used anymore. Let me erase it.

I think that the `sharedInstance` still have its meaning. It is a dependency provided thing, but to accommodate callers, they don't want to log by first asking the logger to the dependencyProvider. So this is shortcut to get the single instance of logger hold by the Criteo singleton.

Reply:
OPT: You could move this shortcut to the macros which are the only callers I guess

Reply:
Tests verify interaction with logger are using it. It's easier for them to just mock the shared logger rather than mocking the shared Criteo, mock a dependency provider, inject the logger, ...